doc/nrf/ug_fw_update.rst

```diff
@@ -0,0 +1,314 @@
+
+#. Program the application to the target development kit and :ref:`check its console output <ug_bootloader_testing>`.
+   With the first firmware version, ``priv_a.pem`` and ``pub_a.pem`` are used for signing and validating the image.
+
+   .. code-block:: console
+
+      *** Booting Zephyr OS build ...  ***
+      Attempting to boot slot 0.
+      Attempting to boot from address 0x9000.
+      Verifying signature against key 0.
+      Hash: 0xda...4f
+      Firmware signature verified.
+      Firmware version 1
+      *** Booting Zephyr OS build ...  ***
+      ...
+
+#. To revoke a single or multiple keys, rebuild the application modifying the configuration setting to use the private key associated with a key listed *after* the currently used key in the list.
+
+   .. code-block:: console
+
+      CONFIG_BUILD_S1_VARIANT=y
+      CONFIG_SB_SIGNING_KEY_FILE="/path/to/priv_c.pem"
+      CONFIG_FW_INFO_FIRMWARE_VERSION=2
+
+   In this example, when compiling with the ``priv_c.pem`` key, images signed with ``priv_a.pem`` or ``priv_b.pem`` will no longer boot when uploaded into an image slot.
+   Also, a firmware version higher than the previous one has been set.
+
+
+#. Deploy the firmware update.
+
+#. Observe the bootloader checking the hashes of the public keys against the new image, then invalidating the earlier keys:
+
+   .. code-block:: console
+
+      *** Booting Zephyr OS build ...  ***
+      Attempting to boot slot 1.
+      Attempting to boot from address 0x84800.
+      Verifying signature against key 0.
+      Hash: 0xda...4f
+      Public key didn't match, try next.
+      Verifying signature against key 1.
+      Hash: 0x5c...f5
+      Public key didn't match, try next.
+      Verifying signature against key 2.
+      Hash: 0x19...73
+      Invalidating key 0.
+      Invalidating key 1.
+      Firmware signature verified.
+      Firmware version 2
+      Setting monotonic counter (version: 2, slot: 1)
+      *** Booting Zephyr OS build ...  ***
+      ...
+
+You can test that the bootloader will no longer boot images signed with the earlier keys by uploading an image signed with one of them.
+
+1. To do this, recompile the application with the following options:
+
+   .. code-block:: console
+
+      CONFIG_SB_SIGNING_KEY_FILE="/path/to/priv_b.pem"
+      CONFIG_FW_INFO_FIRMWARE_VERSION=3
+
+#. To facilitate testing, you can use ``nrfjprog`` to program this image directly into a slot:
+
+   .. code-block:: console
+
+      nrfjprog -f nRF52 -r --verify --program build/zephyr/signed_by_b0_s0_image.hex --sectorerase
+
+#. Observe the bootloader catching the invalid image and booting the valid image in the other slot:
+
+   .. code-block:: console
+
+      *** Booting Zephyr OS build ...  ***
+      Attempting to boot slot 0.
+      Attempting to boot from address 0x9000.
+      Firmware has been invalidated: 0x91020000.
```

Comment:
This indicates that the image has already been invalidated. The output should be something like 

      Key 0 has been invalidated, try next.
      Key 1 has been invalidated, try next.
      Verifying signature against key 2.
      Hash: 0x19...73
      Public key didn't match, try next.
      Failed to validate signature.

Reply:
Was able to reproduce with the output that you referenced. Updated the docs in the latest commit (will be squashed after review).